### PR TITLE
[Misc] Reorg python test according to the guideline.

### PR DESCRIPTION
**Pull request by frozenbugs**
## Description

Reorg python test according to the guideline.
https://docs.google.com/document/d/1djmc7Mtv1h7BVZMFM8kfxq5Z9bzeFcGl-yFkALzdRBU/edit

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: tests/python/common/ops/test_ops.py [hunk 1]
```diff
@@ -290,40 +288,6 @@ def test_sddmm(g, shp, lhs_target, rhs_target, msg, idtype):
         g.edata.pop("m")
 
 
-@pytest.mark.parametrize("g", get_cases(["clique"]))
-@pytest.mark.parametrize("norm_by", ["src", "dst"])
-@pytest.mark.parametrize("shp", edge_softmax_shapes)
-@parametrize_idtype
-def test_edge_softmax(g, norm_by, shp, idtype):
```
Comment (frozenbugs) [inline tests/python/common/ops/test_ops.py:297]: Moved to tests/python/common/ops/test_edge_softmax.py, together with edge_softmax_hetero.

File: tests/python/common/transforms/test_transform.py [hunk 2]
```diff
@@ -28,11 +28,48 @@
 from test_utils.graph_cases import get_cases
 from test_utils import parametrize_idtype
 
-from test_heterograph import create_test_heterograph3, create_test_heterograph4, \
```
Comment (frozenbugs) [inline tests/python/common/transforms/test_transform.py:31]: We should never refer utils from other tests.

File: tests/python/common/backend/test_tensor.py [hunk 3]
```diff
@@ -5,7 +5,6 @@
 
 import dgl
 import dgl.ndarray as nd
-from dgl.utils import toindex
```
Comment (frozenbugs) [inline tests/python/common/backend/test_tensor.py:8]: the whole test is not related to index at all. renamed and removed import.

**Comment by frozenbugs** [comment]
@dgl-bot

**Review by Rhett-Ying — APPROVED** [review]

**Pull request merged**